crates/base-db/src/input.rs

```diff
@@ -270,6 +270,9 @@ pub type TargetLayoutLoadResult = Result<Arc<str>, Arc<str>>;
 pub struct ProcMacro {
     pub name: SmolStr,
     pub kind: ProcMacroKind,
+    /// Ignored proc macros should only be expanded
+    /// when we are explicitly asked to do so by the user.
+    pub ignored: bool,
```

Comment:
Hmm, let's do things a bit differently. Instead of encoding `ignoredness` as a flag here, let's just add a second special `ProcMacroExpander` for ignored ones, samy way we hadnle the dummy expander basically. Then we don't need to fetch the loaded proc-macros everytime to check and can instead immediately check the expander at hand